distribution/ddf-common/src/main/resources-filtered/etc/system.properties

```diff
@@ -49,16 +49,21 @@ javax.net.ssl.trustStoreType=jks
 #
 # Global URL Properties
 #
-# The httpsPort, httpPort, and port properties are used to advertise the system externally
-# The internalHttpsPort and internalHttpPort properties are the ports that the system will run on
+# The httpsPort and httpPort are the ports that the system will run on
 org.codice.ddf.system.protocol=https://
 org.codice.ddf.system.hostname=localhost
```

Comment:
What is the expectation on the CN and alias now in the keystore?  Both match the org.codice.ddf.external.hostname?

Reply:
Both should match the system/internal hostname. The external properties are for a reverse proxy, load balancer, or other independent system that clients are expected to access DDF through.

Reply:
Will that cause problems then?  Will that certificate get exposed all the way out to the consumer?  Will they see browser mismatches?  Or are we expecting the the load balancer or reverse proxy to change out the certificate that gets exposed?

Reply:
In an SSL-terminating configuration the external interface would have its own cert. Trust between the internal interface and DDF would also need to be established.

It is possible to pass certs through in some configurations, which I think could be handled by a SAN for the external interface. Wildcard certs could also work depending on the network topology.

Reply:
DDF does not do wildcard certs well.  And in many environments, we may not have much ownership of the certs.

Reply:
Yeah, we need to do better with wildcard certs.

Reply:
Ah so this is primarily geared towards proxies that will terminate and re-encrypt traffic to the backend then? That might explain why I had to change a couple things in order to get this to work. I had tested by simply forwarding one port to another via `socat`

Are there plans to support raw tcp forwarding better in the future as well?

I think I can probably make it work more easily once these properties are added since i can manually set all of them as well as the linkages to the keystore alias to achieve it, but would be nice to have easier support out of the box. I'm thinking specifically of scenarios where ddf may be running in a container that is exposing a potentially randomized port.

Anyway I'm rambling now, its good to see that we'll be able to support ssl-termination more easily going forward since that was very difficult before

Reply:
I think this would still support TCP forwarding if the certificates were generated correctly. These properties enable DDF to construct the correct URLs, but the certificates still need to be configured correctly.

I think today you have to configure `system.properties` when you randomize the port, so I am not sure how this PR affects that. Are you thinking of picking up the system port more dynamically?

Reply:
Not sure really, currently I have to change the port the whole system runs on since all the urls are relative to the system port. I think if i left the system port on 8993 and changed the external one to 443 while leaving everything else it would probably work with these changes. I think the most ideal situation would be that the urls were generated relative to the incoming request instead since then we wouldn't have to care so much about knowing the external port and hostname. That would probably require quite a lot of changes compared to how we currently do it though.